### DISCUSSION TITLE
FIREBASE PINTEREST: 400 ERROR

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#168] [Help Me!] [answered]
Chris Meffley

No other tickets

 Error: Request failed with status code 400.  Also getting a Cannot read property 'board_title' of null, but I believe the two are related. Since I think that since it's not correctly getting the info from Firebase, then it will be null.

Error messages: see above

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
https://stackoverflow.com/questions/41712562/request-failed-with-status-code-400
https://www.lifewire.com/how-to-fix-a-400-bad-request-error-2617988

All my Promises individually are working, and I'm pretty sure my Promise.all is ok, but if not, I'm not sure where.

domEvents: https://github.com/cmeffley/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/components/events/domEvents.js
Board Promise: https://github.com/cmeffley/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/helpers/data/boardData.js
Pin Promise: https://github.com/cmeffley/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/helpers/data/pinData.js
Promise all: https://github.com/cmeffley/ASSIGNMENT-Firebase-Pinterest/blob/main/src/javascripts/helpers/data/boardPinData.js

**Comment by drteresavasquez** [comment]
@cmeffley what failed with a status of 400? Can you put the whole error here?

Need more details. Also, your google links are not quite what you need to be looking for. Make sure that you fully debug your code and give us those details as well. We need to know what you tried to fix your issue.

For the board title of null, have you console logged the board title or object? What are you getting back?

**Reply by cmeffley** [reply]
I fixed the title null, was missing a / in my url link, I will look into the rest

**Comment by cmeffley** [comment, marked as answer]
Issue 1:
When I did the axios.get for my pins, I was requesting incorrectly. I had ```axios.get(`${dbUrl}/pins/boardId.json``` (or something close to that). But I needed to target the pins more specifically with the ?orderBy:```axios.get(`${dbUrl}/pins.json?orderBy="board_id"&equalTo="${boardId}"`)```

Issue 2:
When I was resolving the promise of getting the specific pins, I did not target the data, and I needed to turn it into an array using: ```resolve(Object.values(response.data)))```

Issue 3: 
In my firebase data I did not have an identifier that matched on the pin and the board. I had just created board_id that linked pins together, but nothing to link the pins to the boards. So I needed to change the pin's board_id to match the board's firebaseKey

Issue 4:
I did not call the correct function to show the pins. I was trying to call the function that got the pins from firebase again.  I did a promise for the pins by themselves, then I called that same promise during the `Promise.all` when, instead, I needed to be showing the pins since I had already gotten them.
